build_tools/python/e2e_test_framework/definitions/iree_definitions.py

```diff
@@ -115,6 +111,15 @@ class MLIRDialectType(Enum):
         MLIRDialectType.MHLO,
 }
 
+# Map to provide stable artificial id of dialect type. Please use and update the
+# next id when adding a new dialect, so we won't reuse the old deprecated id.
+# Next ID: 4
+DIALECT_TYPE_TO_DIALECT_ID_MAP = {
```

Comment:
It seems like we could mixin a dataclass or something here so that the index could get assigned in the enum definition. You could also define __str__ such that it will just stringify as the flag. Anyway, probably too many fancy Python hacks

Reply:
Use dataclass + enum mixin. I want to use `name` for the dialect name but it conflicts with enum's name field.

Reply:
Resolved this first after an offline discussion. Feel free to reopen it